labgrid/driver/rawnetworkinterfacedriver.py

```diff
@@ -0,0 +1,130 @@
+# pylint: disable=no-member
+import contextlib
+import json
+import signal
+import subprocess
+
+import attr
+
+from .common import Driver
+from ..factory import target_factory
+from ..step import step
+from ..util.helper import processwrapper
+from ..util.managedfile import ManagedFile
+from ..resource.common import NetworkResource
+
+
+@target_factory.reg_driver
+@attr.s(eq=False)
+class RawNetworkInterfaceDriver(Driver):
+    bindings = {
+        "iface": {"NetworkInterface", "RemoteNetworkInterface", "USBNetworkInterface"},
+    }
+
+    def __attrs_post_init__(self):
+        super().__attrs_post_init__()
+        self._record_handle = None
+        self._replay_handle = None
+
+    def _wrap_command(self, args):
+        wrapper = ["sudo", "labgrid-raw-interface"]
+
+        if self.iface.command_prefix:
+            # add ssh prefix, convert command passed via ssh (including wrapper) to single argument
+            return self.iface.command_prefix + [" ".join(wrapper + args)]
+        else:
+            # keep wrapper and args as-is
+            return wrapper + args
+
+    def _stop(self, proc, *, timeout=None):
+        assert proc is not None
+
+        try:
+            _, err = proc.communicate(timeout=timeout)
+        except subprocess.TimeoutExpired:
+            proc.send_signal(signal.SIGINT)
```

Comment:
Why do you use `.send_signal(SIGINT)` instead of `.terminate()`? Does tcpdump/tcpreplay have specific behavior for SIGINT?

Reply:
tcpdump does. From tcpdump(1):
> Tcpdump will, if not run with the -c flag, continue capturing packets until it is interrupted by a SIGINT signal (generated, for example, by typing your interrupt character, typically control-C) or a SIGTERM signal (typically generated with the kill(1) command); if run with the -c flag, it will capture packets until it is interrupted by a SIGINT or SIGTERM signal or the specified number of packets have been processed.

Reply:
`.terminate()` sends SIGTERM, so should have the same result. I'd prefer using the standard way of `.terminate()`.

Reply:
Done.